src/provider/systemdmanager/systemdunit.rs

```diff
@@ -212,6 +221,18 @@ impl SystemDUnit {
 
         unit.add_property(Section::Service, "Restart", restart_policy);
 
+        // If `terminationGracePeriodSeconds` was specified in the PodSpec set the value as
+        // 'TimeOutStopSec` on the systemd unit
+        // This means that the service will be killed after this period if it does not shutdown
+        // after receiving a stop command
+        if let Some(stop_timeout) = pod_spec.termination_grace_period_seconds {
+            unit.add_property(
+                Section::Service,
+                "TimeoutStopSec",
+                stop_timeout.to_string().as_str(),
+            );
+        }
```

Comment:
`terminationGracePeriodSeconds` defaults to 30 seconds whereas `DefaultTimeoutStopSpec` is 90 seconds. If no `terminationGracePeriodSeconds` is set then `TimeoutStopSpec` should be set to 30 seconds to match the Kubernetes defaults.

Reply:
True, for some reason I believed that the default setting are a match yesterday..
I've changed this to set the stoptimeout to 30 if nothing is specified in the pod.